src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -523,16 +526,51 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 	private registerListeners(): void {
 
 		// Model Events
-		this._register(this.model.onDidChangeLocked(() => this.onDidChangeGroupLocked()));
-		this._register(this.model.onDidChangeEditorPinned(editor => this.onDidChangeEditorPinned(editor)));
-		this._register(this.model.onDidChangeEditorSticky(editor => this.onDidChangeEditorSticky(editor)));
-		this._register(this.model.onDidMoveEditor(event => this.onDidMoveEditor(event)));
-		this._register(this.model.onDidOpenEditor(editor => this.onDidOpenEditor(editor)));
-		this._register(this.model.onDidCloseEditor(editor => this.handleOnDidCloseEditor(editor)));
-		this._register(this.model.onWillDisposeEditor(editor => this.onWillDisposeEditor(editor)));
-		this._register(this.model.onDidChangeEditorDirty(editor => this.onDidChangeEditorDirty(editor)));
-		this._register(this.model.onDidChangeEditorLabel(editor => this.onDidChangeEditorLabel(editor)));
-		this._register(this.model.onDidChangeEditorCapabilities(editor => this.onDidChangeEditorCapabilities(editor)));
+		this._register(this.model.onDidModelChange(e => {
+			this._onDidModelChange.fire(e);
+			if (e.kind === GroupChangeKind.GROUP_LOCKED) {
+				this.onDidChangeGroupLocked();
+				return;
+			}
+			if (!e.editor) {
+				return;
+			}
+			switch (e.kind) {
+				case GroupChangeKind.EDITOR_PIN:
+					this.onDidChangeEditorPinned(e.editor);
+					break;
+				case GroupChangeKind.EDITOR_STICKY:
+					this.onDidChangeEditorSticky(e.editor);
+					break;
+				case GroupChangeKind.EDITOR_MOVE:
+					if (e.oldEditorIndex !== undefined && e.editorIndex !== undefined) {
```

Comment:
I wonder if it would be easier to introduce overloads of this model event that makes some of the event properties mandatory when a certain change kind is set.

Reply:
This would be ideal. I will have to explore how to do that in typescript. I wonder if it's possible with a union type

Reply:
Pseudo code:

```ts
interface MyEvent {
    kind: string;
    meta?: boolean;
}

interface MySpecificEvent extends MyEvent {
    kind: 'change';
    meta: boolean;
}
```

Reply:
I made this change but typescript doesn't seem smart enough to infer the stronger type. I'm not sure how it would know what type to pick though because the function would be `IGenericGroupEvent | IMoveEvent | ICloseEvent` and then `kind: 'move'` would still be able to satisfy both `IGenericGroupEvent | IMoveEvent` meaning we still would need a type guard or a cast